libs/shared/types/src/lib/types.ts

```diff
@@ -6,9 +8,63 @@ export enum EventStatus {
   ERROR = 'ERROR',
 }
 
-export type S3Source = string;
+export type S3Source = { inputType: InputType; value: string };
```

Comment:
We want inputType to be something like `InputType.S3` or something, any input type cannot be here, it should be the specific one for the source.

Same for all of them in this file.